### ISSUE TITLE
Better proxy support

### ISSUE THREAD

**Issue by alanhamlett** [wakatime/wakatime#23]
Currently proxy support is handled with [requests](http://docs.python-requests.org/en/latest/user/advanced/#proxies), but only simple HTTP Basic Auth is supported.

Support for other proxy schemes is needed, such as domain auth:

`https://domain\user:pass@address:port`

**Comment by alanhamlett** [comment]
Dependent on kennethreitz/requests#1582

**Comment by alanhamlett** [comment]
Possibly could use [requests/requests-ntlm](https://github.com/requests/requests-ntlm).

**Comment by rs38** [comment]
since 2(?) weeks I cannot upload wakatimes through a proxy, 

the log show errors like this:

```
{"now": "2016/05/17 18:03:52 +0200", "version": "6.0.2", "plugin": "visualstudio/12.0 visualstudio-wakatime/8.0.3", "time": 1463501013537000.0, "level": "ERROR", "message": {"response_code": 400, "response_content": "{\n  \"errors\": {\n    \"time\": [\n      \"Number must be between 
1388534400 and 99999999999.\"\n    ]\n  }\n}"}}
{"now": "2016/05/17 18:13:50 +0200", "version": "6.0.2", "plugin": "visualstudio/12.0 visualstudio-wakatime/8.0.3", "time": 1463501561156857.0, "level": "ERROR", "message": {"response_content": "{\n  \"errors\": {\n    \"time\": [\n      \"Number must be between 1388534400 and 99999999999.\"\n    ]\n  }\n}", "response_code": 400}}
{"now": "2016/05/17 18:28:30 +0200", "version": "6.0.2", "plugin": "visualstudio/12.0 visualstudio-wakatime/8.0.3", "time": 1463502466650297.0, "level": "ERROR", "message": {"response_code": 400, "response_content": "{\n  \"errors\": {\n    \"time\": [\n      \"Number must be between 1388534400 and 99999999999.\"\n    ]\n  }\n}"}}

   raise SSLError(e, request=request)\nwakatime.packages.requests.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)\n", "SSLError": "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)"}}

```

**Comment by alanhamlett** [comment]
That's not a proxy problem, that's a bug in the Visual Studio plugin version that you're using. Please upgrade WakaTime in Visual Studio to the latest version 8.0.4 to fix it.

**Comment by rs38** [comment]
this is 8.0.4:

`[WakaTime Info 05:18:42 PM] Initializing WakaTime v8.0.4
[WakaTime Debug 05:18:43 PM] Python found from embedded location: C:\Users\x\AppData\Roaming\WakaTime\python\pythonw
[WakaTime Info 05:18:43 PM] Current wakatime-cli version is 6.0.2
[WakaTime Info 05:18:43 PM] Checking for updates to wakatime-cli...
[WakaTime Debug 05:18:43 PM] A proxy with authentication will be used.
[WakaTime Info 05:18:44 PM] wakatime-cli is up to date.
[WakaTime Info 05:18:44 PM] Finished initializing WakaTime v8.0.4`

**Comment by gandarez** [comment]
@rs38 How did you setup wakatime config proxy?

**Comment by alanhamlett** [comment]
@rs38 Your first message shows `"plugin": "visualstudio/12.0 visualstudio-wakatime/8.0.3",` but now that you're using v8.0.4 is it working?

**Comment by rs38** [comment]
no it still does not work. Not sure when it updated.
Will contact Support. thx.
(at home with out Proxy it works fine, okay different VS)

**Comment by rs38** [comment]
@gandarez http(s)://user:passw@proxy:81
tried both http/https

**Comment by alanhamlett** [comment]
Latest master adds support for NTLM proxies, anyone have a corporate network they could test this out on?

**Comment by alanhamlett** [comment]
Fixed with v6.2.3.
